### PR TITLE
52 view to display all cards returned by the api

### PR DESCRIPTION
**Pull request by Bjolidon**
Implement the view for the web Api. It is not ready to be merged as I need to add some more tests.

### CODE REVIEW

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/browser/BrowserApiFragment.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import com.google.gson.Gson
+
+
+/**
+ * Fragment that displays the cards returned by the API in a recycler view
+ */
+class BrowserApiFragment : Fragment() {
+    private var _binding: FragmentBrowserApiBinding? = null
+    private lateinit var viewModel: BrowserApiViewModel
+    private val binding get() = _binding!!
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/sdp/tarjetakuna/ui/browser/BrowserApiFragment.kt:23]: Where is this used?

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/browser/UserCollectionFragment.kt [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+ * Fragment that displays the cards in a recycler view
+ */
+class UserCollectionFragment : Fragment() {
+
+    private var _binding: FragmentUserCollectionBinding? = null
+    private lateinit var viewModel: UserCollectionViewModel
+
+    // This property is only valid between onCreateView and
+    // onDestroyView.
+    private val binding get() = _binding!!
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/sdp/tarjetakuna/ui/browser/UserCollectionFragment.kt:29]: Also unused?

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/browser/BrowserFragmentTest.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+
+@RunWith(AndroidJUnit4::class)
+class BrowserFragmentTest {
+
+    private lateinit var activityRule: ActivityScenario<MainActivity>
```
Comment (Cocograal) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/browser/BrowserFragmentTest.kt:31]: Did you merge main into your branch? if so the location should make these tests fail no?
Consider adding the GrantedPermissionRule (you can find these in the tests for the Location)

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragmentUITest.kt [hunk 4]
```diff
@@ -76,56 +79,108 @@ class WebApiFragmentUITest {
...
+            } catch (e: Error) {
+                // Assertion failed, continue looping until timeout
+                Espresso.onIdle()
+            }
+        }
 
-        // check that the text change to the response
-        onView(withId(R.id.api_results)).check(matches(withSubstring("Marsh Goblins")))
+        // check that the test passed
+        assert(result)
```
Comment (Cocograal) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragmentUITest.kt:105]: You should use assertThat and not assert

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragmentUITest.kt [hunk 5]
```diff
@@ -76,56 +79,108 @@ class WebApiFragmentUITest {
...
+                // Assertion failed, continue looping until timeout
+                Espresso.onIdle()
+            }
+        }
 
-        // click on the button
-        onView(withId(R.id.api_sets_button)).perform(waitForText(R.string.api_sets, 200))
-        onView(withId(R.id.api_sets_button)).perform(click())
+        // check that the test passed
+        assert(result)
```
Comment (Cocograal) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragmentUITest.kt:144]: here as well

**Review by Cocograal — CHANGES_REQUESTED** [review]
Nice jobs! Be sure you use the right assertions in testing

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/browser/BrowserFragmentTest.kt (hunk 3 above)
Comment (Bjolidon) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/browser/BrowserFragmentTest.kt:31]: Yes the branch is up to date with main but I can add the given permission to be sure it keeps working.

**Review by Cocograal — APPROVED** [review]
Thanks for the changes!

**Pull request merged**